Tutorials/Tutorial27_PostProcessing/assets/shaders/ComputeLighting.fx

```diff
@@ -0,0 +1,140 @@
+#include "BasicStructures.fxh"
+#include "FullScreenTriangleVSOutput.fxh"
+
+#define M_PI                      3.14159265358979
+#define M_EPSILON                 1e-3
+
+#define FLT_EPS                   5.960464478e-8
+#define FLT_MAX                   3.402823466e+38
+#define FLT_MIN                   1.175494351e-38
+
+cbuffer cbCameraAttribs
+{
+    CameraAttribs g_Camera;
+}
+
+Texture2D<float3> g_TextureBaseColor;
+Texture2D<float3> g_TextureMaterialData;
+Texture2D<float4> g_TextureNormal;
+Texture2D<float>  g_TextureDepth;
+Texture2D<uint>   g_TextureObjectID;
+Texture2D<float4> g_TextureSSR;
+
+TextureCube<float3> g_TextureEnvironmentMap;
+SamplerState        g_TextureEnvironmentMap_sampler;
+
+TextureCube<float3> g_TextureIrradianceMap;
+SamplerState        g_TextureIrradianceMap_sampler;
+
+TextureCube<float3> g_TexturePrefilteredEnvironmentMap;
+SamplerState        g_TexturePrefilteredEnvironmentMap_sampler;
+
+Texture2D<float2> g_TextureBRDFIntegrationMap;
+SamplerState      g_TextureBRDFIntegrationMap_sampler;
+
+struct SurfaceInformation
+{
+    float3 BaseColor;
+    float Roughness;
+    float Metallness;
+    float Occlusion;
+    float3 Normal;
+    float3 Position;
+};
+
+float3 ComputeWorldPosition(float2 NormalizedXY, float Depth)
+{
+    float4 NDC = float4(NormalizedXY, DepthToNormalizedDeviceZ(Depth), 1.0f);
+    float4 Position = mul(NDC, g_Camera.mViewProjInv);
+    Position /= Position.w;
+    return Position.xyz;
+}
+
+float3 FresnelSchlickRoughness(float CosTheta, float3 F0, float roughness)
+{
+    float Alpha = 1.0 - roughness;
+    return F0 + (max(float3(Alpha, Alpha, Alpha), F0) - F0) * pow(clamp(1.0 - CosTheta, 0.0, 1.0), 5.0);
+}
+
+float3 SampleEnvironmentMap(float3 Coord)
+{
+    return g_Camera.f4ExtraData[0].y * g_TextureEnvironmentMap.Sample(g_TextureEnvironmentMap_sampler, float3(+1.0, -1.0, +1.0) * Coord);
```

Comment:
Please save `g_Camera.f4ExtraData[0].y` to a separate variable for clarity to understand what this value stores.